compiler/rustc_middle/src/middle/privacy.rs

```diff
@@ -56,7 +55,35 @@ impl EffectiveVisibility {
     }
 
     pub fn is_public_at_level(&self, tag: AccessLevel) -> bool {
-        self.get(tag).map_or(false, |vis| vis.is_public())
+        self.get(tag).is_public()
+    }
+
+    fn update(&mut self, vis: Visibility<DefId>, tag: AccessLevel, tree: impl DefIdTree) -> bool {
+        let mut changed = false;
+        for level in [
+            AccessLevel::Public,
+            AccessLevel::Exported,
+            AccessLevel::Reachable,
+            AccessLevel::ReachableFromImplTrait,
+        ] {
```

Comment:
The order doesn't matter in this specific loop, or in the loop below.
We could add a static method that returns this array to `AccessLevel`.

Reply:
It would be nice to compile time assert that that array contains all `AccessLevel` variants like https://docs.rs/strum_macros/0.24.3/strum_macros/derive.EnumIter.html, but this crate will be overkill.

Reply:
Yeah, anything complex would be an overkill here, it's highly unlikely that this enum will be extended (we should rather try to remove `ReachableFromImplTrait` instead).